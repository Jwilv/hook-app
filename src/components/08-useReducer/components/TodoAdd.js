import React from 'react'
import { useForm } from '../../../hooks/useForm'

export const TodoAdd = ({handleAddTodo}) => {

    const [{ description }, handleInputChanGet, reset] = useForm({
        description: ''
    })

    const handleSubmit = (event) => {
        event.preventDefault();

        if (description.trim().length <= 1) return;

        const newTodo = {
            id: new Date().getTime(),
            desc: description,
            done: false,
        }

        handleAddTodo(newTodo);
        reset();
    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type='text'
                name='description'
                placeholder='Incluir...'
                autoComplete='off'
                className='form-control'
                value={description}
                onChange={handleInputChanGet}
            />
            <button
                type='submit'
                className='btn btn-block btn-outline-primary mt-1 btn-block'
            >
                Agregar
            </button>
        </form>
    )
}
