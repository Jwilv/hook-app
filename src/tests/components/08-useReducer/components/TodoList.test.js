import React from 'react'
import { shallow } from 'enzyme'
import { TodoList } from '../../../../components/08-useReducer/components/TodoList';
import { demoTodo } from '../../../fixtures/demoTodos';


describe('pruebas en el componente <TodoList />', () => {

    const handleToggle = jest.fn();
    const handleDelete = jest.fn();

    const wrapper = shallow(<TodoList
        todos={demoTodo}
        handleToggle={handleToggle}
        handleDelete={handleDelete}
    />)

    test('debe de mostrarse correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de renderizar todos los todo <TodoListItem /> ', () => {
        expect(wrapper.find("TodoListItem").length).toBe(demoTodo.length)

    });

    test('se debe de enviar 2 funciones handleToggle /  handleDelete', () => {
        expect(wrapper.find("TodoListItem").at(0).prop('handleToggle')).toEqual( expect.any(Function) )
        expect(wrapper.find("TodoListItem").at(0).prop('handleDelete')).toEqual( expect.any(Function) )
    });
})