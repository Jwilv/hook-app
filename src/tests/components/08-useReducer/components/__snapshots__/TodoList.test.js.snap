// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente <TodoList /> debe de   1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={0}
    key="1"
    todo={
      Object {
        "desc": "aprender a jugar",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={1}
    key="2"
    todo={
      Object {
        "desc": "al minecraft",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;

exports[`pruebas en el componente <TodoList /> debe de mostrarse correctamente  1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={0}
    key="1"
    todo={
      Object {
        "desc": "aprender a jugar",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={1}
    key="2"
    todo={
      Object {
        "desc": "al minecraft",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;

exports[`pruebas en el componente <TodoList /> debe de tener   1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={0}
    key="1"
    todo={
      Object {
        "desc": "aprender a jugar",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={1}
    key="2"
    todo={
      Object {
        "desc": "al minecraft",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;

exports[`pruebas en el componente <TodoList /> debe de tener 2 <  1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={0}
    key="1"
    todo={
      Object {
        "desc": "aprender a jugar",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={1}
    key="2"
    todo={
      Object {
        "desc": "al minecraft",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;

exports[`pruebas en el componente <TodoList /> debe de tener 2 <T  1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={0}
    key="1"
    todo={
      Object {
        "desc": "aprender a jugar",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={1}
    key="2"
    todo={
      Object {
        "desc": "al minecraft",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;

exports[`pruebas en el componente <TodoList /> debe de tener 2 <TodoList  1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={0}
    key="1"
    todo={
      Object {
        "desc": "aprender a jugar",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={1}
    key="2"
    todo={
      Object {
        "desc": "al minecraft",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;

exports[`pruebas en el componente <TodoList /> debe de tener 2 <TodoListItem  1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={0}
    key="1"
    todo={
      Object {
        "desc": "aprender a jugar",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={1}
    key="2"
    todo={
      Object {
        "desc": "al minecraft",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;

exports[`pruebas en el componente <TodoList /> debe de tener 2 <TodoListItem />  1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={0}
    key="1"
    todo={
      Object {
        "desc": "aprender a jugar",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={1}
    key="2"
    todo={
      Object {
        "desc": "al minecraft",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;

exports[`pruebas en el componente <TodoList /> debe de tener 2 todo list   1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={0}
    key="1"
    todo={
      Object {
        "desc": "aprender a jugar",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={1}
    key="2"
    todo={
      Object {
        "desc": "al minecraft",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
